File: server/coffeeshop/src/main/java/com/hust/coffeeshop/services/CustomerService.java
package com.hust.coffeeshop.services;

import com.hust.coffeeshop.models.dto.PagingListResponse;
import com.hust.coffeeshop.models.dto.customer.CustomerFilterRequest;
import com.hust.coffeeshop.models.dto.customer.CustomerRequest;
import com.hust.coffeeshop.models.dto.customer.CustomerResponse;
import org.springframework.stereotype.Service;

public interface CustomerService {
    //api tạo
    CustomerResponse create(CustomerRequest request);

    //api update
    CustomerResponse update(CustomerRequest request, int id);

    //api xóa
    void delete(int id);

    //api get id
    CustomerResponse getById(int id);

    //api filter
    PagingListResponse<CustomerResponse> filter(CustomerFilterRequest filter) ;
}

File: client/coffee-shop/src/services/CustomerService/CustomerService.ts
import axios, { AxiosResponse } from "axios";
import { getAxiosConfig } from "../config";
import { CustomerFilterRequest, CustomerRequest, CustomerResponse, ListCustomerResponse } from "./types";

class CustomerService {
    static async filter(filter?: CustomerFilterRequest): Promise<AxiosResponse<ListCustomerResponse>> {
        return axios.get(`/customers`, { ...getAxiosConfig(), params: filter});
    }
    static async create(request: CustomerRequest): Promise<AxiosResponse<CustomerResponse>> {
        return axios.post(`/customers`,request , { ...getAxiosConfig()});
    }
    static async update(request: CustomerRequest, id?: number): Promise<AxiosResponse<CustomerResponse>> {
        return axios.put(`/customers/${id}`,request, { ...getAxiosConfig()});
    }
    static async getById(id: string): Promise<AxiosResponse<CustomerResponse>> {
        return axios.get(`/customers/${id}`, { ...getAxiosConfig()});
    }
    static async delete(id?: number): Promise<AxiosResponse<CustomerResponse>> {
        return axios.delete(`/customers/${id}`, { ...getAxiosConfig()});
    }
}
export default CustomerService;